src/ppom-middleware.test.ts

```diff
@@ -0,0 +1,89 @@
+import { storageBackendReturningData } from '../test/test-utils';
+import { PPOMController } from './ppom-controller';
+import { createPPOMMiddleware } from './ppom-middleware';
+
+jest.mock('./ppom.ts', () => ({
+  PPOM: class PPOMClass {
+    #jsonRpcRequest;
+
+    constructor(jsonRpcRequest: any) {
+      this.#jsonRpcRequest = jsonRpcRequest;
+    }
+
+    validateJsonRpc = async () => {
+      return Promise.resolve();
+    };
+
+    free = () => undefined;
+
+    testJsonRPCRequest = () => this.#jsonRpcRequest();
+  },
+  ppomInit: () => undefined,
+}));
+
+describe('createPPOMMiddleware', () => {
+  it('should return PPOM Middleware function', () => {
+    const ppomController = new PPOMController({
+      storageBackend: storageBackendReturningData,
+      provider: { sendAsync: Promise.resolve() },
+      chainId: '0x1',
+      onNetworkChange: (_callback) => undefined,
+    });
+    const middlewareFunction = createPPOMMiddleware(ppomController);
+    expect(middlewareFunction).toBeDefined();
+  });
+
+  describe('PPOMMiddleware', () => {
```

Comment:
I'm not sure I follow what this grouping is, if all these tests are verifying the logic of `createPPOMMiddleware`.

Reply:
good point, let me re-group